storage/pin/pin.go

```diff
@@ -115,24 +168,25 @@ func (p *PinAPI) PinFiles(rootHash string, isRaw bool, credentials string) error
 // that are encountered on the way. The pre-requisite is that the file should
 // have been already pinned using the PinFiles function. This function can
 // be called only from an external command.
-func (p *PinAPI) UnpinFiles(rootHash string, credentials string) error {
-
+func (p *API) UnpinFiles(rootHash string, credentials string) error {
 	addr, err := hex.DecodeString(rootHash)
 	if err != nil {
-		log.Error("Error decoding root hash", "rootHash", rootHash, "Reason", err.Error())
+		log.Error("Error decoding root hash", "rootHash", rootHash, "err", err)
 		return err
 	}
 
-	isRawInDB, err := p.db.IsPinnedFileRaw(chunk.Address(addr))
-	if err != nil {
-		log.Error("Root hash is not pinned", "rootHash", rootHash, "Reason", err.Error())
+	p.pinLock.RLock()
+	fileInfo, ok := p.pinnedFiles[rootHash]
+	p.pinLock.RUnlock()
+	if !ok {
+		log.Error("Root hash is not pinned", "rootHash", rootHash)
```

Comment:
here you return err is incorrect. 
Best to construct and error 

```
if  !ok {
    return errors.New("not found")
}

Reply:
Refactored the code.. this returns proper error now.